lib/kvbm-logical/CLAUDE.md

```
@@ -0,0 +1,71 @@
+cargo build -p kvbm-logical --features testing
+```
+
+## Architecture
+
+**kvbm-logical** is the core logical block lifecycle manager for KVBM (KV Block Manager). It manages KV cache blocks for LLM inference through a type-safe state machine, registry, and pool system.
+
+### Block Lifecycle (Type-State Pattern)
+
+Blocks use compile-time type states to enforce valid transitions:
+
+```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)